### PR TITLE
DeviceInputSystem: Add handling for pointercancel events

### PR DESCRIPTION
**Pull request by PolygonalSun**
This PR contains additional event handling for `pointercancel` events.  The event factory was also modified to now assign a value for pointerType.

### CODE REVIEW

**Comment by RaananW** [comment]
When is pointercancel triggered in the case of babylon's element? by pointerout scenarios?

File: src/DeviceInput/Helpers/eventFactory.ts [hunk 1]
```diff
@@ -50,6 +50,8 @@ export class DeviceEventFactory {
 
         evt.pointerId = deviceType === DeviceType.Mouse ? 1 : deviceSlot;
 
+        evt.pointerType = deviceType === DeviceType.Mouse ? "mouse" : "touch";
```
Comment (sebavan) [inline src/DeviceInput/Helpers/eventFactory.ts:53]: nit: can we enclose in a if with the previous line ?

**Comment by RaananW** [comment]
> pointercancel events are triggered when the OS or some process interrupts what the pointer is doing and takes over the pointer, with respect to context. For example, on mobile, if you are trying to move things in the canvas and initiate a "Back" gesture, a pointercancel event will fire. On Desktop, it could be something like clicking and dragging an image on the page. It should be noted that with Mouse, these events are incredibly rare.
> 
> If a pointercancel event is fired, a pointerout and pointerleave will also be fired as a result afterwards but a pointerout will should never trigger a pointercancel.

So the general idea is to release all locked pointers and trigger a pointer up if the pointer state was "on", right?

LGTM, seb left a comment.

File: src/DeviceInput/Helpers/eventFactory.ts (hunk 1 above)
Comment (PolygonalSun) [reply src/DeviceInput/Helpers/eventFactory.ts]: Fixed

**Review by sebavan — APPROVED** [review]
LGTM, let s address @RaananW comments :-)

**Comment by PolygonalSun** [comment]
> > pointercancel events are triggered when the OS or some process interrupts what the pointer is doing and takes over the pointer, with respect to context. For example, on mobile, if you are trying to move things in the canvas and initiate a "Back" gesture, a pointercancel event will fire. On Desktop, it could be something like clicking and dragging an image on the page. It should be noted that with Mouse, these events are incredibly rare.
> > If a pointercancel event is fired, a pointerout and pointerleave will also be fired as a result afterwards but a pointerout will should never trigger a pointercancel.
> 
> So the general idea is to release all locked pointers and trigger a pointer up if the pointer state was "on", right?
> 
> LGTM, seb left a comment.

Yup, that's pretty much the idea.  It's kind of like a mini-blur.

**Pull request merged**